server/src/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -1104,14 +1107,61 @@ public void cleanupStorage(boolean recurring) {
                     }
 
                     // remove snapshots in Error state
-                    List<SnapshotVO> snapshots = _snapshotDao.listAllByStatus(Snapshot.State.Error);
+                    List<SnapshotVO> snapshots = _snapshotDao.listAllByStatusIncludingRemoved(Snapshot.State.Error);
                     for (SnapshotVO snapshotVO : snapshots) {
                         try {
                             List<SnapshotDataStoreVO> storeRefs = _snapshotStoreDao.findBySnapshotId(snapshotVO.getId());
+                            boolean isVMware = snapshotVO.getHypervisorType().equals(HypervisorType.VMware);
```

Comment:
@DaanHoogland @mike-tutkowski Partial disks are left behind in VMware env when SSVM is shutdown/destroyed. So, `DeleteSnapshotsDirCommand` for Image store and `DeleteVMSnapshotCommand` for Primary store (to remove worker VM snapshot) has been triggered in VMware env here to cleanup the partial disks created in secondary storage and VM snapshots in primary respectively. The VMware storage manager would cleanup the worker VM snapshot from `DeleteVMSnapshotCommand`. No VMware specific code here, except sending these commands for cleanup.